### PR TITLE
get annotation(s)

### PR DESCRIPTION
**Pull request by Maxwell-Yang-PDFTron**
Functions:
getAnnotationAtPoint
getAnnotationsAtLine
getAnnotationsOnPage

### CODE REVIEW

File: ios/RNTPTDocumentView.m [hunk 1]
```diff
@@ -3018,6 +3018,90 @@ - (void)setPropertiesForAnnotation:(NSString *)annotationId pageNumber:(NSIntege
...
+    
+    __block NSDictionary *annotation;
+    if (pdfViewCtrl && pdfDoc) {
+        NSError *error;
+        
+        [pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+            PTAnnot *annot = [pdfViewCtrl GetAnnotationAt:(int)x y:(int)y distanceThreshold:distanceThreshold minimumLineWeight:minimumLineWeight];
+            
+            if (annot && [annot IsValid]) {
+                annotation = [self getAnnotationData:annot pageNumber:[[annot GetPage] GetIndex] pdfViewCtrl:pdfViewCtrl];
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:3036]: `[annot GetPage]` is not reliable and should not be used. Instead, you should use https://www.pdftron.com/api/ios/Classes/PTPDFViewCtrl.html#/c:objc(cs)PTPDFViewCtrl(im)GetPageNumberFromScreenPt:y: to get the page number. (and use https://www.pdftron.com/api/ios/Classes/PTPDFDoc.html#/c:objc(cs)PTPDFDoc(im)GetPage: if you ever need the page object.)

File: ios/RNTPTDocumentView.m [hunk 2]
```diff
@@ -3018,6 +3018,90 @@ - (void)setPropertiesForAnnotation:(NSString *)annotationId pageNumber:(NSIntege
...
+    __block NSMutableArray *annotations = [[NSMutableArray alloc] init];
+    if (pdfViewCtrl && pdfDoc) {
+        NSError *error;
+        
+        [pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+            NSArray <PTAnnot *> *annots = [pdfViewCtrl GetAnnotationListAt:(int)x1 y1:(int)y1 x2:(int)x2 y2:(int)y2];
+            
+            for (PTAnnot *annot in annots) {
+                if ([annot IsValid]) {
+                    [annotations addObject:[self getAnnotationData:annot pageNumber:[[annot GetPage] GetIndex] pdfViewCtrl:pdfViewCtrl]];
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:3063]: Do not use `[annot GetPage]` as explained above.

File: ios/RNTPTDocumentView.m [hunk 3]
```diff
@@ -3018,6 +3018,90 @@ - (void)setPropertiesForAnnotation:(NSString *)annotationId pageNumber:(NSIntege
...
+    __block NSMutableArray *annotations = [[NSMutableArray alloc] init];
+    if (pdfViewCtrl && pdfDoc) {
+        NSError *error;
+        
+        [pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+            NSArray <PTAnnot *> *annots = [pdfViewCtrl GetAnnotationsOnPage:(int)pageNumber];
+            
+            for (PTAnnot *annot in annots) {
+                if ([annot IsValid]) {
+                    [annotations addObject:[self getAnnotationData:annot pageNumber:[[annot GetPage] GetIndex] pdfViewCtrl:pdfViewCtrl]];
```
Comment (JamesTron) [inline ios/RNTPTDocumentView.m:3091]: Do not use `[annot GetPage]` as explained above.

**Review by JamesTron — CHANGES_REQUESTED** [review]

File: API.md [hunk 4]
```diff
@@ -1444,6 +1444,87 @@ this._viewer.setPropertiesForAnnotation('Pdftron', 1, {
...
+
+```js
+this._viewer.getAnnotationAtPoint(167, 287, 100, 10).then((annotation) => {
+  if (annotation) {
+    console.log('Annotation found at point (167, 287) has id:', annotation.id);
+  }
+})
+```
+
+#### getAnnotationsAtLine
```
Comment (sgong-pdftron) [inline API.md:1475]: bad API name, it should be getAnnotationAt (one) and getAnnotationListAt (multiple), it could be a line, a rect etc.

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 5]
```diff
@@ -2432,6 +2445,92 @@ public void deleteAnnotations(ReadableArray annots) throws PDFNetException {
...
+
+        if (pdfViewCtrl != null && doc != null) {
+            boolean shouldUnlockRead = false;
+            try {
+                doc.lockRead();
+                shouldUnlockRead = true;
+                Annot annot = pdfViewCtrl.getAnnotationAt(x, y, distanceThreshold, minimumLineWeight);
+
+                if (annot != null && annot.isValid()) {
+                    annotation = getAnnotationData(annot, annot.getPage().getIndex());
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:2462]: never use getPage, see James comment for this, apply to all usage of getPage

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Review by JamesTron — APPROVED** [review]

**Pull request merged**